src/terminal/input/mouseInput.cpp

```diff
@@ -322,98 +318,80 @@ bool TerminalInput::HandleMouse(const til::point position,
+            {
+                return _GenerateUtf8Sequence(position, realButton, isHover, modifierKeyState, delta);
+            }
+            else if (_inputMode.test(Mode::SgrMouseEncoding))
+            {
+                // For SGR encoding, if no physical buttons were pressed,
+                // then we want to handle hovers with WM_MOUSEMOVE.
+                // However, if we're dragging (WM_MOUSEMOVE with a button pressed),
+                //      then use that pressed button instead.
+                return _GenerateSGRSequence(position, physicalButtonPressed ? realButton : button, _isButtonDown(realButton), isHover, modifierKeyState, delta);
+            }
+            else
             {
-                std::wstring sequence;
-                if (_inputMode.test(Mode::Utf8MouseEncoding))
-                {
-                    sequence = _GenerateUtf8Sequence(position,
-                                                     realButton,
-                                                     isHover,
-                                                     modifierKeyState,
-                                                     delta);
-                }
-                else if (_inputMode.test(Mode::SgrMouseEncoding))
-                {
-                    // For SGR encoding, if no physical buttons were pressed,
-                    // then we want to handle hovers with WM_MOUSEMOVE.
-                    // However, if we're dragging (WM_MOUSEMOVE with a button pressed),
-                    //      then use that pressed button instead.
-                    sequence = _GenerateSGRSequence(position,
-                                                    physicalButtonPressed ? realButton : button,
-                                                    _isButtonDown(realButton), // Use realButton here, to properly get the up/down state
-                                                    isHover,
-                                                    modifierKeyState,
-                                                    delta);
-                }
-                else
-                {
-                    sequence = _GenerateDefaultSequence(position,
-                                                        realButton,
-                                                        isHover,
-                                                        modifierKeyState,
-                                                        delta);
-                }
-                success = !sequence.empty();
-
-                if (success)
-                {
-                    _SendInputSequence(sequence);
-                    success = true;
-                }
-                if (_inputMode.any(Mode::ButtonEventMouseTracking, Mode::AnyEventMouseTracking))
-                {
-                    _mouseInputState.lastPos.x = position.x;
-                    _mouseInputState.lastPos.y = position.y;
-                    _mouseInputState.lastButton = button;

```

Comment:
Didn't this get lost in the change?

Reply:
This code change is a little (very) confusing tbh. This if condition has moved up in the scop, so that I could use early returns and remove the local `success` variable. It's fine to move the `if` condition up because none of the called functions depend on `_mouseInputState` (they're all 3 `static` functions).